tensorflow-ops/tensorflow-ops.cabal

```diff
@@ -16,18 +16,31 @@ library
   exposed-modules: TensorFlow.Gradient
                  , TensorFlow.Ops
                  , TensorFlow.EmbeddingOps
+                 , TensorFlow.Logging
   build-depends:  proto-lens == 0.1.*
                 , base >= 4.7 && < 5
                 , bytestring
-                , fgl
-                , mtl
+                , conduit
```

Comment:
As you noted, having heavy dependencies in the core will slow things down. I vote for introducing tensorflow-logging, unless you believe everything will have logging.